Extension/src/LanguageServer/client.ts

```diff
@@ -948,13 +1028,22 @@ export class DefaultClient implements Client {
 
                     if (isFirstClient) {
                         workspaceReferences = new refs.ReferencesManager(this);
-
                         // The configurations will not be sent to the language server until the default include paths and frameworks have been set.
                         // The event handlers must be set before this happens.
-                        const inputCompilerDefaults: configs.CompilerDefaults = await languageClient.sendRequest(QueryCompilerDefaultsRequest, {});
+                        const compilerPath: PersistentState<string[]> = new PersistentState<string[]>("CPP.trustedCompilerPaths", []);
+                        const inputCompilerDefaults: configs.CompilerDefaults = await this.requestCompiler(compilerPath.Value);
                         compilerDefaults = inputCompilerDefaults;
                         this.configuration.CompilerDefaults = compilerDefaults;
-
+                        this.configuration.setupConfigurations();
+                        clientCount++;
+                        // count number of clients, once all clients are configured, check for trusted compiler to display notification to user and add a short delay to account for config provider logic to finish
+                        if (clientCount === vscode.workspace.workspaceFolders?.length) {
+                            if (!compilerDefaults.trustedCompilerFound && !defaultCompilerFound) {
+                                // if there is no compilerPath in c_cpp_properties.json, prompt user to configure a compiler
+                                this.promptSelectCompiler(false);
+                                defaultCompilerFound = true;
```

Comment:
The call to `promptSelectCompiler` is not awaited.  That is actually good, as we don't want to stall the whole extension while waiting for the user to respond to that UI.  However, that also means we don't yet know if a default compiler was found.  Maybe `defaultCompilerFound` should not be set here?  Or, if your intention was to use this boolean as the `I already did this check` boolean from my other comment, I'd suggest naming it more concisely.

Reply:
Do you have a recommendation for what it should be called?